src/app/services/widgetservice/widget.service.ts

```diff
@@ -0,0 +1,33 @@
+import {Injectable} from '@angular/core';
+import {WidgetUiMode} from "../../widget/WidgetUiMode";
+import {WeatherWidget} from "../../interfaces/weatherwidget";
+import {interval, startWith} from "rxjs";
+import {map, switchMap} from "rxjs/operators";
+import {WeatherService} from "../weatherservice/weather.service";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WidgetService {
+  constructor(private weatherService: WeatherService){}
+  serviceData(widget:WeatherWidget){
+    interval(3000)
+      .pipe(
+        startWith(0),
+        switchMap(() => this.weatherService.getWeather(widget.name)),
+        map((data) => new WidgetUiMode(data))
+      ).
```

Comment:
return this.weatherService.getWeather(widget.name).pipe( map((data) => new WidgetUiMode(data)))